crates/consensus/authority/src/block_fetcher.rs

```diff
@@ -189,43 +190,42 @@ where
                 botanix_consensus_pkg.clone(),
                 self.evm_config.clone(),
                 &self.client,
+                false,
             ) {
                 Ok(bundle_state) => {
                     let senders =
                         TransactionSigned::recover_signers(&block.body, block.body.len()).unwrap();
                     let sealed_block_with_senders =
                         SealedBlockWithSenders::new(sealed_block.clone(), senders)
                             .expect("senders are valid");
-                    // Process Botanix specific logs
-                    // get pegouts if btc_server is available
-                    // only federation nodes will have btc_server
-                    let mut pegouts = match self.btc_server.as_ref() {
-                        Some(btc_server) => {
-                            let pegouts = match crate::utils::process_receipts(
-                                &mut btc_server.clone(),
-                                &bundle_state,
-                                bitcoin_block_header.1,
-                                self.btc_network,
-                                self.consensus.chain_spec.parent_confirmation_depth,
-                            )
-                            .await
-                            {
-                                Ok(pegouts) => pegouts,
-                                Err(e) => {
-                                    error!(target: "consensus::authority", ?e, "Failed to process botanix log");
-                                    continue;
-                                }
-                            };
+                    let header = sealed_block.header.clone();
 
-                            pegouts
+                    // process pegins
+                    // must be done before getting utxo commitment
+                    let rpc = self.btc_server.as_mut().expect("btc_server exists");
```

Comment:
```suggestion
                    let btc_signing_server_rpc = self.btc_server.as_mut().expect("btc_server exists");
```

More verbose is fine here

Reply:
done!